### ISSUE TITLE
pry-style leveled indent?

### ISSUE THREAD

**Issue by mpyw** [bobthecow/psysh#234]
Will this feature be available in the future?

### Current

```ShellSession
example@localhost:~$ psysh
Psy Shell v0.5.2 (PHP 7.0.0-dev — cli) by Justin Hileman
>>> function () {
... function () {
... function () {
```

### Future

```ShellSession
example@localhost:~$ psysh
Psy Shell v0.5.2 (PHP 7.0.0-dev — cli) by Justin Hileman
>>> function () {
...   function () {
...     function () {
```

I also wish the handler `SIGINT` were overridden not to exit shell but to skip current line, just like most Ruby and Python shells.

**Comment by bobthecow** [comment]
The indent would be a bit difficult, as we can't really parse an incomplete statement, so we can't really tell anything but the fact that the previous line is continued. We could tokenize the code, and build our own partial AST to decide how indented something should be, but (so far) that has felt like more trouble than it's worth.

We do accept pull requests, though :)

**Comment by nowox** [comment]
I totally agree about the (<kbd>CTRL</kbd>+<kbd>C</kbd>) `SIGINT` action. As for most shells, it is the `EOF` (End Of File) which is used to close the session with  (<kbd>CTRL</kbd>+<kbd>D</kbd>).

**Comment by bobthecow** [comment]
This is available [in the latest release](https://github.com/bobthecow/psysh/releases/tag/v0.12.21) (via the `--experimental-readline` flag). Try it out and let me know how it goes!
